### PR TITLE
Migrate ChefSpec to InSpec

### PR DESCRIPTION
**Pull request by khashf**

### CODE REVIEW

**Comment by khashf** [comment]
Worked around issue on `InSpec` doesn't support including `share_example.rb` in other tests:
* Move the `share_example.rb` to `tests/integration/helpers/`
* Remove appearances of reference to `share_examples.rb` in other tests
* For each test that include `share_examples.rb`, in the same directory level, create a symlink that **relatively** points to the location of `share_example.rb`. That is, if you are standing at `tests/integration/default/inspec/`, then the symlink in that directory should be `../../helpers/share_example.rb`

**Comment by khashf** [comment]
Related issue at InSpec repo: https://github.com/inspec/inspec/issues/1522

File: test/integration/helpers/shared_examples.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+if os[:release].to_i < 7
```
Comment (ramereth) [inline test/integration/helpers/shared_examples.rb]: While this is OK, I'd rather we use the InSpec described method using the [os resource](https://www.inspec.io/docs/reference/resources/os/) and change it to the following:
```suggestion
if os.release.to_i < 7
```

Comment (khashf) [reply test/integration/helpers/shared_examples.rb]: I've committed the above change suggestion, but I couldn't find where else I need to change based on your comment of "use the InSpec described method using the os resource". Could you elaborate a little bit more?

File: test/integration/helpers/shared_examples.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
     it { should be_running }
   end
 end
-describe command('rpcinfo -p localhost') do
+describe bash('rpcinfo -p localhost') do
```
Comment (ramereth) [inline test/integration/helpers/shared_examples.rb]: I'd prefer to keep this using the ``command`` resource unless we absolutely need to run something using the bash interpreter.

File: .kitchen.yml [hunk 3]
```diff
@@ -6,3 +6,5 @@ suites:
   - name: homes
     run_list:
       - recipe[osl-nfs::homes]
+verifier:
+  name: inspec
```
Comment (ramereth) [inline .kitchen.yml]: I'd prefer this to be at the top.

**Pull request merged**